### DISCUSSION TITLE
not working on macos

### DISCUSSION THREAD

**Discussion by danilo-patrucco** [iann0036/iamlive#86] [Q&A]
Hello, 

I'm trying to run iamlive in a Macos running sonoma 14.4.1 without much success, the command I'm running is:

`iamlive --set-ini --profile dev --output-file policy.json --refresh-rate 1 --sort-alphabetical --host 127.0.0.1 --mode csm` 
I set the profile using `export AWS_PROFILE=dev` then run iamlive and then run `terraform apply "test.out"` on a separate terminal, but the policy has the following output:

```zsh 
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": null,
            "Resource": "*"
        }
    ]
}
```
before running the I am command I also run this:

`export AWS_CSM_ENABLED=true`

the shell I'm using is zsh.

Is there something I'm missing?

**Comment by danilo-patrucco** [comment]
By adding the variables 
``` 
export AWS_CSM_ENABLED=true
export AWS_CSM_PORT=31000
export AWS_CSM_HOST=127.0.0.1
``` 
in the terraform terminal I was able to get an output, but it was a single output, I am running modules called from a different repository, should it be able to read calls made through sequential modules (I am calling 6-7 modules from the main.tf file)

**Reply by iann0036** [reply]
Hi @danilo-patrucco,

There might be limited CSM coverage for the specific actions you are running, or Terraform's internal SDK is not cooperating as it should.

Perhaps try moving to proxy mode by:

* Using `--mode proxy` for the iamlive command in one terminal; and
* Setting the following environment variables in your Terraform terminal just prior to running:
```
export HTTP_PROXY=http://127.0.0.1:10080
export HTTPS_PROXY=http://127.0.0.1:10080
export AWS_CA_BUNDLE=~/.iamlive/ca.pem
```

**Reply by danilo-patrucco** [reply]
Thank you @iann0036, I will give it a shot and let you know the results, I appreciate the time you take in answering my question.
